### PR TITLE
Feature/evaluation function

### PR DESCRIPTION
**Pull request by JWG-1**
added a Function that only evaluates the already trained model

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -427,6 +440,7 @@ def fit(self):
 	parser.add_argument('-seed',            dest='seed',            default=41504,  type=int,     	help='Seed for randomization')
 
 	parser.add_argument('-restore',         dest='restore',         action='store_true',            help='Restore from the previously saved model')
+	parser.add_argument('-evaluate', 		dest='evaluate', 		action='store_true', 			help='Whether evaluate the model')
```
Comment (A7exSchin) [inline run.py]: help='Only evaluating a given model, model name must be available in the checkpoints folder'

Comment (JWG-1) [reply run.py]: fixed

File: Readme.md [hunk 2]
```diff
@@ -103,6 +103,9 @@ We evaluated the CompGCN repository with the following configurations.
   
   ##### Overall BEST:
   python run.py -name best_model -score_func conve -opn corr 
+  
+  #### Evaluate an already trained Model:
+  python run.py -name XXX -score_func conve -opn corr
```
Comment (EhrenbruderRoschtbraten) [inline Readme.md]: Your introduced parameter -evaluate is missing there.
```suggestion
  python run.py -name XXX -score_func conve -opn corr -evaluate
```

**Review by EhrenbruderRoschtbraten — CHANGES_REQUESTED** [review]
Thanks for implementing that, but it didn't work for me, because when trying to reevaluate the trained conve model from the logs and models repo with the following command, i get the following error.
Command: 
```batchfile
python run.py -name ConvE_Sub_codex_s_no_GCN_model -score_func conve -opn sub -data codex-s -evaluate
```
Error:
```
Traceback (most recent call last):
  File "C:\Daten\Hochschule\Wintersemester 22-23\Forschungsarbeit\CompGCN\run.py", line 478, in <module>
    model.evaluteOnly()
  File "C:\Daten\Hochschule\Wintersemester 22-23\Forschungsarbeit\CompGCN\run.py", line 373, in evaluteOnly
    self.load_model(save_path)
  File "C:\Daten\Hochschule\Wintersemester 22-23\Forschungsarbeit\CompGCN\run.py", line 261, in load_model
    self.model.load_state_dict(state_dict)
  File "C:\Daten\Hochschule\Wintersemester 22-23\Forschungsarbeit\CompGCN\venv\lib\site-packages\torch\nn\modules\module.py", line 1671, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for CompGCN_ConvE:
        Missing key(s) in state_dict: "conv1.w_loop", "conv1.w_in", "conv1.w_out", "conv1.w_rel", "conv1.loop_rel", "conv1.bn.weight", "conv1.bn.bias", "conv1.bn.running_mean", "conv1.bn.running_var".
        size mismatch for bn2.weight: copying a param with shape torch.Size([100]) from checkpoint, the shape in current model is torch.Size([200]).
        size mismatch for bn2.bias: copying a param with shape torch.Size([100]) from checkpoint, the shape in current model is torch.Size([200]).
        size mismatch for bn2.running_mean: copying a param with shape torch.Size([100]) from checkpoint, the shape in current model is torch.Size([200]).
        size mismatch for bn2.running_var: copying a param with shape torch.Size([100]) from checkpoint, the shape in current model is torch.Size([200]).
        size mismatch for m_conv1.weight: copying a param with shape torch.Size([200, 1, 5, 5]) from checkpoint, the shape in current model is torch.Size([200, 1, 7, 7]).
        size mismatch for fc.weight: copying a param with shape torch.Size([100, 19200]) from checkpoint, the shape in current model is torch.Size([200, 39200]).
        size mismatch for fc.bias: copying a param with shape torch.Size([100]) from checkpoint, the shape in current model is torch.Size([200]).
```
Do we need to input all parameters for the trained model?

**Comment by JWG-1** [comment]
> 

Yes u need to set all the parameters the same as when u trained the model

**Comment by JWG-1** [comment]
> 
> ```batchfile
> python run.py -name ConvE_Sub_codex_s_no_GCN_model -score_func conve -opn sub -data codex-s -evaluate
> ```

you forgot do disable the gnn encoder

File: run.py [hunk 3]
```diff
@@ -362,7 +362,19 @@ def run_epoch(self, epoch, val_mrr = 0):
 		loss = np.mean(losses)
 		self.logger.info('[Epoch:{}]:  Training Loss:{:.4}\n'.format(epoch, loss))
 		return loss
+	def evaluteOnly(self):
```
Comment (1Maxnet1) [inline run.py:368]: Could you please add a newline before the function and name it evaluate_only according to paython method naming guidelines

**Review by 1Maxnet1 — COMMENTED** [review]
Looks good to me, except for my one comment. I sadly wasn't able to execute it, but due to a local error on my side

**Comment by 1Maxnet1** [comment]
> Looks good to me, except for my one comment. I sadly wasn't able to execute it, but due to a local error on my side

I made it work locally on my machine, except for the comment it looks good

File: run.py (hunk 3 above)
Comment (JWG-1) [reply run.py:368]: fixed

**Review by 1Maxnet1 — APPROVED** [review]
Looks good to me :+1:  Thanks

**Comment by A7exSchin** [comment]
Evaluating works + pritns more performance metrics. All good to me

**Review by A7exSchin — APPROVED** [review]

**Pull request merged**